Source/Core/WebService.swift

```diff
@@ -181,21 +217,24 @@ extension WebService {
      Return an absolute URL string relative to the baseURLString value.
     
      - parameter string: URL string.
-     - returns: An absoulte URL string relative to the value of `baseURLString`.
+     - returns: An absolute URL string relative to the value of `baseURLString`.
     */
-    public func absoluteURLString(_ string: String) -> String {
-        return constructURLString(string, relativeToURLString: baseURLString)
+    dynamic public func absoluteURLString(_ string: String) -> String {
+        return constructURLString(string, relativeToURL: baseURL)
     }
-    
+
     /**
      Return an absolute URL string relative to the baseURLString value.
     
      - parameter string: URL string value.
      - parameter relativeURLString: Value of relative URL string.
      - returns: An absolute URL string.
     */
-    func constructURLString(_ string: String, relativeToURLString relativeURLString: String) -> String {
-        let relativeURL = URL(string: relativeURLString)
-        return URL(string: string, relativeTo: relativeURL)!.absoluteString
+    func constructURLString(_ string: String, relativeToURL: URL?) -> String {
```

Comment:
Now that `Request` can be initialized with a URL type, we should change `absoluteURLString` and `constructURLString` to `absoluteURL` and `constructURL` and have them return URL types instead of strings. The only concern I have about changing `absoluteURLString` is that it is a public API and although it is probably rarely used, we could easily support backward compatibility by changing the implementation to just call `absoluteURL` and return an absolute string.

This way we are only converting to URL strings when explicitly calling one of the string property or methods instead of converting between types through the flow of request creation.